### PR TITLE
Invalidate session after successful user creation

### PR DESCRIPTION
**Pull request by ganesanarun**

### CODE REVIEW

File: src/main/java/in/projecteka/consentmanager/user/SignUpService.java [hunk 1]
```diff
@@ -22,21 +23,14 @@
 import java.util.concurrent.ExecutionException;
 import java.util.function.Function;
 
+@AllArgsConstructor
 public class SignUpService {
 
     private final static Logger logger = Logger.getLogger(SignUpService.class);
+    public static final long JWT_TOKEN_VALIDITY = 5 * 60 * 60;
```
Comment (angshu) [inline src/main/java/in/projecteka/consentmanager/user/SignUpService.java:30]: if its duration, then this should not be hard coded. Should be loaded from application property.

File: src/main/java/in/projecteka/consentmanager/user/SignUpService.java [hunk 2]
```diff
@@ -22,21 +23,14 @@
...
     private JWTProperties jwtProperties;
     private LoadingCache<String, Optional<String>> unverifiedSessions;
```
Comment (angshu) [inline src/main/java/in/projecteka/consentmanager/user/SignUpService.java:32]: Why not use Spring cache Abstraction instead of using Guava cache directly?

Comment (angshu) [reply src/main/java/in/projecteka/consentmanager/user/SignUpService.java:32]: Guava cache is also in memory cache, right? and does not work in a cluster.

File: src/main/java/in/projecteka/consentmanager/user/SignUpService.java (hunk 1 above)
Comment (ganesanarun) [reply src/main/java/in/projecteka/consentmanager/user/SignUpService.java]: Done.

File: src/main/java/in/projecteka/consentmanager/user/SignUpService.java (hunk 2 above)
Comment (ganesanarun) [reply src/main/java/in/projecteka/consentmanager/user/SignUpService.java]: Apparently Spring inbuilt cache does not support value expiry by default, but guava cache does that, that's why Abhisek used that.

Yeah, we could use the abstraction, but not a mandatory now though.

**Comment by ganesanarun** [comment]
@angshu can you review please?

File: src/main/java/in/projecteka/consentmanager/user/SignUpService.java (hunk 2 above)
Comment (angshu) [reply src/main/java/in/projecteka/consentmanager/user/SignUpService.java]: ok. I will put that in our tech debt list.

**Pull request merged**